### ISSUE TITLE
Get 'unable to find vcvarsall.bat when 'import surprise'

### ISSUE THREAD

**Issue by bigheadming** [spyder-ide/spyder#5087]
## Description
Get 'unable to find vcvarsall.bat when trying to run 'import surprise'. Package surprise is a python package which was installed successfully by 'pip install numpy' and 'pip install scikit-surprise'. I have tried running 'import surprise' under ipython and jupyter-notebook in the same machine, both run successfully. However, 'import surprise' gives me  'unable to find vcvarsall.bat when 'import surprise' in spyder. 

**What steps will reproduce the problem?**

1. (This step is for installing a package called 'surprise') pip install numpy, pip install scikit-surprise (see  https://github.com/NicolasHug/Surprise)
2. Launch spyder. 
3. Either in an editor or directly in ipython console, type 'import surprise'. Then the 'unable to find vcvarsall.bat when 'import surprise' appears.

**What is the expected output? What do you see instead?**
I expect the command goes okay. But I got the followings: 

import surprise
Traceback (most recent call last):

  File "<ipython-input-1-ca7cc839673c>", line 1, in <module>
    import surprise

  File "C:\ProgramData\Anaconda2\lib\site-packages\surprise\__init__.py", line 3, in <module>
    from .prediction_algorithms import AlgoBase

  File "C:\ProgramData\Anaconda2\lib\site-packages\surprise\prediction_algorithms\__init__.py", line 22, in <module>
    from .algo_base import AlgoBase

  File "C:\ProgramData\Anaconda2\lib\site-packages\surprise\prediction_algorithms\algo_base.py", line 10, in <module>
    from .. import similarities as sims

  File "C:\ProgramData\Anaconda2\lib\site-packages\pyximport\pyximport.py", line 458, in load_module
    language_level=self.language_level)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pyximport\pyximport.py", line 233, in load_module
    exec("raise exc, None, tb", {'exc': exc, 'tb': tb})

  File "C:\ProgramData\Anaconda2\lib\site-packages\pyximport\pyximport.py", line 215, in load_module
    inplace=build_inplace, language_level=language_level)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pyximport\pyximport.py", line 191, in build_module
    reload_support=pyxargs.reload_support)

  File "C:\ProgramData\Anaconda2\lib\site-packages\pyximport\pyxbuild.py", line 102, in pyx_to_dll
    dist.run_commands()

  File "C:\ProgramData\Anaconda2\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)

  File "C:\ProgramData\Anaconda2\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()

  File "C:\ProgramData\Anaconda2\lib\site-packages\Cython\Distutils\old_build_ext.py", line 185, in run
    _build_ext.build_ext.run(self)

  File "C:\ProgramData\Anaconda2\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()

  File "C:\ProgramData\Anaconda2\lib\site-packages\Cython\Distutils\old_build_ext.py", line 193, in build_extensions
    self.build_extension(ext)

  File "C:\ProgramData\Anaconda2\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)

  File "C:\ProgramData\Anaconda2\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()

  File "C:\ProgramData\Anaconda2\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)

  File "C:\ProgramData\Anaconda2\lib\distutils\msvc9compiler.py", line 271, in query_vcvarsall
    raise DistutilsPlatformError("Unable to find vcvarsall.bat")

ImportError: Building module surprise.similarities failed: ['DistutilsPlatformError: Unable to find vcvarsall.bat\n']

**Please provide any additional information below**
I installed spyder via Anaconda (for python 2.7), hence got ipython and jupyter-notebook as well. All running fine with 'import surprise', including launching python via cmd. However, spyder is the only one that has the problem. I have already installed MS Visual C++ compiler for Python 2.7.

## Version and main components

* Spyder Version: 3.2.1 
* Python Version: 2.7.13
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
IPython >=4.0;<6.0:  5.3.0 (OK)
cython >=0.21     :  0.26 (OK)
jedi >=0.9.0      :  0.10.2 (OK)
nbconvert >=4.0   :  5.2.1 (OK)
numpy >=1.7       :  1.13.1 (OK)
pandas >=0.13.1   :  0.20.3 (OK)
pycodestyle >=2.3 :  2.3.1 (OK)
pyflakes >=0.5.0  :  1.6.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
pylint >=0.25     :  1.7.2 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
rope >=0.9.4      :  0.9.4 (OK)
sphinx >=0.6.6    :  1.6.3 (OK)
sympy >=0.7.3     :  1.1.1 (OK)

```

**Comment by ccordoba12** [comment]
Did you start Spyder from a system terminal (cmd.exe) before running this?

**Comment by bigheadming** [comment]
I started with the icon created by Anaconda when installing it. I am tried to start from a cmd.exe and type in 'spyder' (as you suggested). Then type in 'import surprise'. Both the same error. 

 

    On Wednesday, August 30, 2017 1:23 AM, Carlos Cordoba <notifications@github.com> wrote:
 

 Did you start Spyder from a system terminal (cmd.exe) before running this?—

**Comment by ccordoba12** [comment]
No idea why this is happening, sorry. We'll try to take a look at this in a future release.

**Comment by bigheadming** [comment]
Thanks for your reply. Please take a look if possible. It's very annoying that I have to switch to other IDE when using that package (python package called surprise) as I am working on some project that I need to use it very often. This is especially true when Spyder 3.2.1 only supports ipython console now (in the past, I could choose to use python console to avoid this problem.) Thanks. 

    On Wednesday, August 30, 2017 9:20 AM, Carlos Cordoba <notifications@github.com> wrote:
 

 No idea why this is happening, sorry. We'll try to take a look at this in a future release.—

**Comment by MatevzKu** [comment]
Same issue here with surpriselib as well
Traceback (most recent call last):

  File "<ipython-input-1-bfee19a548d3>", line 1, in <module>
    runfile('E:/Dropbox_mk/Dropbox (Personal)/PyCode/Testbed/surp.py', wdir='E:/Dropbox_mk/Dropbox (Personal)/PyCode/Testbed')

  File "C:\Anaconda\lib\site-packages\spyder\utils\site\sitecustomize.py", line 710, in runfile
    execfile(filename, namespace)

  File "C:\Anaconda\lib\site-packages\spyder\utils\site\sitecustomize.py", line 86, in execfile
    exec(compile(scripttext, filename, 'exec'), glob, loc)

  File "E:/Dropbox_mk/Dropbox (Personal)/PyCode/Testbed/surp.py", line 1, in <module>
    from surprise import SVD

  File "C:\Anaconda\lib\site-packages\surprise\__init__.py", line 3, in <module>
    from .prediction_algorithms import AlgoBase

  File "C:\Anaconda\lib\site-packages\surprise\prediction_algorithms\__init__.py", line 22, in <module>
    from .algo_base import AlgoBase

  File "C:\Anaconda\lib\site-packages\surprise\prediction_algorithms\algo_base.py", line 10, in <module>
    from .. import similarities as sims

  File "C:\Anaconda\lib\site-packages\pyximport\pyximport.py", line 458, in load_module
    language_level=self.language_level)

  File "C:\Anaconda\lib\site-packages\pyximport\pyximport.py", line 233, in load_module
    exec("raise exc, None, tb", {'exc': exc, 'tb': tb})

  File "C:\Anaconda\lib\site-packages\pyximport\pyximport.py", line 215, in load_module
    inplace=build_inplace, language_level=language_level)

  File "C:\Anaconda\lib\site-packages\pyximport\pyximport.py", line 191, in build_module
    reload_support=pyxargs.reload_support)

  File "C:\Anaconda\lib\site-packages\pyximport\pyxbuild.py", line 102, in pyx_to_dll
    dist.run_commands()

  File "C:\Anaconda\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)

  File "C:\Anaconda\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()

  File "C:\Anaconda\lib\site-packages\Cython\Distutils\old_build_ext.py", line 185, in run
    _build_ext.build_ext.run(self)

  File "C:\Anaconda\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()

  File "C:\Anaconda\lib\site-packages\Cython\Distutils\old_build_ext.py", line 193, in build_extensions
    self.build_extension(ext)

  File "C:\Anaconda\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)

  File "C:\Anaconda\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()

  File "C:\Anaconda\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)

  File "C:\Anaconda\lib\distutils\msvc9compiler.py", line 271, in query_vcvarsall
    raise DistutilsPlatformError("Unable to find vcvarsall.bat")

ImportError: Building module surprise.similarities failed: ['DistutilsPlatformError: Unable to find vcvarsall.bat\n']

the interesting thing is that runing the same script from the command line (outside of spyder) works without any problem.

**Comment by ccordoba12** [comment]
This is a duplicate of issue #5299.

**Comment by jmrichardson** [comment]
I am getting this error too.  Looked at the duplicate issue but don't know how to fix?

**Comment by ccordoba12** [comment]
@jmrichardson, we're working on a fix (#6607). It'll be available on our 3.2.8 version.
